### PR TITLE
1.21.4 support

### PR DESCRIPTION
**Pull request by booky10**
All relevant changes I've collected: https://haste.pinofett.de/3b5icru3x4.yml

This PR should be pretty much done and has been brievely tested on a 1.21.4-rc3 fabric server - more tests should be done after other 1.21.4 platforms have been released

### CODE REVIEW

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java [hunk 1]
```diff
@@ -47,9 +48,15 @@ public enum Action {
...
         /**
-         * Updates the order in which the player is listed in the tablist.
+         * Updates the order in which the player is listed in the tablist.<br>
+         * Added with 1.21.2.
          */
-        UPDATE_LIST_ORDER;
+        UPDATE_LIST_ORDER,
+        /**
+         * Updates wether the outer skin layer (hat) of the player will be shown in tablist.<br>
+         * Added with 1.21.5.
```
Comment (Cyramek) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java]: 1.21.4?

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java [hunk 2]
```diff
@@ -63,7 +70,14 @@ public static class PlayerInfo {
         private Component displayName;
         @Nullable
         private RemoteChatSession chatSession;
-        private int listOrder; // added in 1.21.2
+        /**
+         * Added with 1.21.2
+         */
+        private int listOrder;
+        /**
+         * Added with 1.21.5
```
Comment (Cyramek) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java]: 1.21.4?

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java [hunk 3]
```diff
@@ -126,10 +151,20 @@ public GameMode getGameMode() {
...
 
+        /**
+         * Added with 1.21.2
+         */
         public int getListOrder() {
             return this.listOrder;
         }
 
+        /**
+         * Added with 1.21.5
```
Comment (Cyramek) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java]: 1.21.4?

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java [hunk 4]
```diff
@@ -154,9 +189,19 @@ public void setChatSession(@Nullable RemoteChatSession chatSession) {
...
 
+        /**
+         * Added with 1.21.2
+         */
         public void setListOrder(int listOrder) {
             this.listOrder = listOrder;
         }
+
+        /**
+         * Added with 1.21.5
```
Comment (Cyramek) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java]: 1.21.4?

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java (hunk 1 above)
Comment (retrooper) [reply api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerPlayerInfoUpdate.java]: These errors have been resolved. Thanks.

**Comment by retrooper** [comment]
Thanks for the PR.

**Pull request merged**